mobile/src/board/hooks/useBoard.ts

```diff
@@ -0,0 +1,438 @@
+      [3, [45, 38, 31, 24, 17, 10, 3]],
+      [5, [47, 40, 33, 26, 19, 12, 5]],
+    ]),
+  ],
+  [
+    Direction.Left,
+    new Map([
+      [1, [7, 8, 9, 10, 11, 12, 13]],
+      [3, [21, 22, 23, 24, 25, 26, 27]],
+      [5, [35, 36, 37, 38, 39, 40, 41]],
+    ]),
+  ],
+]);
+
+enum Action {
+  RotateRemaining = "ROTATE_REMAINING",
+  GameStateChange = "GAME_STATE_CHANGE",
+  PlaceTile = "PLACE_TILE",
+  PlayerTurnChange = "PLAYER_TURN_CHANGE",
+}
+
+enum RotationDirection {
+  Clockwise = "CLOCKWISE",
+  AntiClockwise = "ANTICLOCKWISE",
+}
+
+interface RotationPayload {
+  direction: RotationDirection;
+  rotation: Rotation;
+}
+
+interface PlaceTilePayload {
+  direction: Direction;
+  index: number;
+}
+
+interface MovePawnPayload {
+  line: number;
+  row: number;
+}
+
+type GameStateChangePayload = GameState;
+
+type NewPlayerTurnPayload = null;
+
+interface UserAction {
+  kind: Action;
+  payload:
+    | RotationPayload
+    | PlaceTilePayload
+    | MovePawnPayload
+    | GameStateChangePayload
+    | NewPlayerTurnPayload;
+}
+
+export function useBoard(id: number | string): [Board | null, any | null] {
+  const [board, setBoard] = useState<Board | null>(null);
+  const [error, setError] = useState<any | null>(null);
+
+  const setGameState = (gameState: GameState) => {
+    setBoard((board) =>
+      board
+        ? {
+            ...board,
+            gameState,
+          }
+        : null
+    );
+  };
+
+  const fetchBoard = async function () {
+    try {
+      const updatedBoard = await boardRepository.getById(id);
+      setBoard(updatedBoard);
+    } catch (e) {
+      setError("Failed to load board");
+      setBoard(null);
+    }
+  };
+
+  const handleMercureMessage = async ({ data }: { data: string }) => {
+    const actions: UserAction[] = JSON.parse(data);
+
+    for (const { kind, payload } of actions) {
+      const resolver = actionResolvers.get(kind);
+      if (resolver) {
+        await resolver(payload);
+      }
+    }
+  };
+
+  const updateTiles = async (
+    direction: Direction,
+    { top, left }: { top?: number; left?: number },
+    index: number
+  ) => {
+    await animationFrame(() => {
+      const tileIndexes = moveTileIndexes.get(direction)!.get(index)!;
+      const lastIndex = tileIndexes[6];
+
+      setBoard((board) => {
+        if (!board) {
+          return null;
+        }
+
+        const { state } = board;
+        const { tiles, remainingTile } = state;
+
+        return {
+          ...board,
+          state: {
+            ...state,
+            tiles: tiles.map((tileLine, line) => {
+              return tileLine.map((boardTile, row) => {
+                const index = line * 7 + row;
+                if (!tileIndexes.includes(index)) {
+                  return boardTile;
+                }
+
+                return {
+                  ...boardTile,
+                  top,
+                  left,
+                  opacity: index == lastIndex ? 0 : 1,
```

Comment:
When you compare, it's a good habit to do a strict compare `index === lastIndex` . The difference between `==` and `===` is that is also compare type.
```
1 == "1" => true
1 === "1" => false
```